packages/cli/src/modules/mcp/mcp.oauth.controller.ts

```diff
@@ -0,0 +1,74 @@
+import { authorizationHandler } from '@modelcontextprotocol/sdk/server/auth/handlers/authorize.js';
+import { clientRegistrationHandler } from '@modelcontextprotocol/sdk/server/auth/handlers/register.js';
+import { revocationHandler } from '@modelcontextprotocol/sdk/server/auth/handlers/revoke.js';
+import { tokenHandler } from '@modelcontextprotocol/sdk/server/auth/handlers/token.js';
+import { Get, RootLevelController, StaticRouterMetadata } from '@n8n/decorators';
+import { Container } from '@n8n/di';
+import type { Response, Request, Router } from 'express';
+
+import { UrlService } from '@/services/url.service';
+
+import { McpOAuthService, SUPPORTED_SCOPES } from './mcp-oauth-service';
+
+const mcpOAuthService = Container.get(McpOAuthService);
+
+@RootLevelController('/')
+export class McpOAuthController {
+	constructor(private readonly urlService: UrlService) {}
+
+	static routers: StaticRouterMetadata[] = [
+		{
+			path: '/mcp-oauth/register',
+			router: clientRegistrationHandler({ clientsStore: mcpOAuthService.clientsStore }) as Router,
+			skipAuth: true,
+		},
+		{
+			path: '/mcp-oauth/authorize',
+			router: authorizationHandler({ provider: mcpOAuthService }) as Router,
+			skipAuth: true,
+		},
+		{
+			path: '/mcp-oauth/token',
+			router: tokenHandler({ provider: mcpOAuthService }) as Router,
+			skipAuth: true,
+		},
+		{
+			path: '/mcp-oauth/revoke',
+			router: revocationHandler({ provider: mcpOAuthService }) as Router,
+			skipAuth: true,
+		},
```

Comment:
Does `@modelcontextprotocol/sdk` add rate limiter for these? If not, we should add our own rate limiter